src/test/java/uk/gov/hmcts/reform/pcq/commons/utils/JsonFeignResponseUtilTest.java

```diff
@@ -0,0 +1,200 @@
+        try {
+            pcqRecordWithoutCaseResponseOptional = JsonFeignResponseUtil.decode(response,
+                    PcqRecordWithoutCaseResponse.class);
+        } catch (IOException e) {
+            log.error("IOException occurred {} ", e.getMessage());
+            fail("Not expected to get IO Exception here");
+        } finally {
+            response.close();
+        }
+
+
+        assertThat(pcqRecordWithoutCaseResponseOptional).isNotEmpty();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testDecode_fails_with_ioException() {
+        Map<String, Collection<String>> header = new ConcurrentHashMap<>();
+        Collection<String> list = new ArrayList<>();
+        header.put(ENCODING_STR, list);
+
+        Response.Body bodyMock = mock(Response.Body.class);
+        Response response = Response.builder().status(200).reason("OK").headers(header).body(bodyMock).request(
+                mock(Request.class)).build();
+
+        try {
+            when(bodyMock.asInputStream()).thenThrow(new IOException());
+            when(bodyMock.asReader(UTF_8)).thenThrow(new IOException());
+            bodyMock.close();
+        } catch (IOException e) {
+            log.error("Error during execution {}", e.getMessage());
+        }
+
+        Optional<Object> createUserProfileResponseOptional = Optional.empty();
+        try {
+            createUserProfileResponseOptional = JsonFeignResponseUtil.decode(response,
+                    PcqRecordWithoutCaseResponse.class);
+        } catch (IOException e) {
+            log.error("Error during execution {}", e.getMessage());
+        } finally {
+            response.close();
+        }
+
+        assertThat(createUserProfileResponseOptional).isEmpty();
+
+
+    }
+
+    @Test
+    public void test_convertHeaders() {
+        Map<String, Collection<String>> header = new ConcurrentHashMap<>();
+        Collection<String> list = new ArrayList<>(Arrays.asList("gzip", "request-context", "x-powered-by",
+                "content-length"));
+        header.put(ENCODING_STR, list);
+
+        MultiValueMap<String, String> responseHeader = JsonFeignResponseUtil.convertHeaders(header);
+        assertThat(responseHeader).isNotEmpty();
+
+        Collection<String> emptylist = new ArrayList<>();
+        header.put(ENCODING_STR, emptylist);
+        MultiValueMap<String, String> responseHeader1 = JsonFeignResponseUtil.convertHeaders(header);
+
+        assertThat(responseHeader1.get(ENCODING_STR)).isEmpty();
+    }
+
+    @Test
+    public void test_toResponseEntity_with_payload_not_empty() {
+        Map<String, Collection<String>> header = new ConcurrentHashMap<>();
+        Collection<String> list = new ArrayList<>(Arrays.asList("a", "b"));
+        header.put(ENCODING_STR, list);
+
+        Response response = Response.builder().status(200).reason("OK").headers(header).body(
+                "{"
```

Comment:
This JSON looks the same as previous line 43, also if this isn't coming from a file can it be made a static final string?

Reply:
Agreed. Created a static final string.